protocol/rpcconsumer/relay_processor.go

```diff
@@ -48,12 +48,11 @@ type RelayProcessor struct {
 	protocolResponseErrors       RelayErrors
 	successResults               []common.RelayResult
 	lock                         sync.RWMutex
-	chainMessage                 chainlib.ChainMessage
+	protocolMessage              chainlib.ProtocolMessage
 	guid                         uint64
 	selection                    Selection
 	consumerConsistency          *ConsumerConsistency
-	dappID                       string
-	consumerIp                   string
+	userData                     chainlib.UserData
```

Comment:
userData is never set, but it is referenced, also it shouldn't exist as a member, should always reference through protocolmessage.GetUserData, and the fact we didn't capture it in unitests means we dont have any checking this, can you add?